code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -818,6 +818,8 @@
 				updateicon()
 			else
 				to_chat(user, "<span class='warning'>Access denied.</span>")
+	else if(isEmag(W))
+		emag_check(W,user)
```

Comment:
Is this a necessary check? I believe the item_attack.dm check may cover this case, but can't test it right now (heading to work).
```suggestion
```

Reply:
mob attackby doesn't call or supercall any of this so it's here